framework/src/wifi_manager/wifi_manager.c

```diff
@@ -0,0 +1,477 @@
+#include <slsi_wifi/slsi_wifi_api.h>
+#include <semaphore.h>
+#include "wifi_utils.h"
+#include "wifi_mutex.h"
+#include "wifi_net.h"
+
+static void *g_dhcp_handle = NULL;
+static dhcp_sta_joined g_new_sta_join = NULL;
+#ifdef CONFIG_ENABLE_IOTIVITY
+mqd_t g_dw_nwevent_mqfd;
+#endif
+
+/*
+ * Iotivity stack receives network events from wifi manager module
+ * this will be changed later
+ */
+#ifdef CONFIG_ENABLE_IOTIVITY
+void __tizenrt_manual_linkset(const char *msg)
+{
+	int ret = mq_send(g_dw_nwevent_mqfd, msg, 3, 42);
+	if(ret < 0) {
+		ndbg("send message fail\n");
+		return;
+	}
+}
+#endif
+
+static wifi_manager_info g_manager_info;
+
+static wifi_mutex w_mutex;
+
+static wifi_softap_config *g_wifi_softap_config;
+
+static wifi_utils_result start_dhcp_client(void)
+{
+	struct dhcpc_state state;
+	int ret;
+
+	g_dhcp_handle = dhcpc_open(CTRL_IFNAME);
+	if (g_dhcp_handle == NULL) {
+		ndbg("Invalid dhcp handle\n");
+		return WIFI_UTILS_FAIL;
+	}
+	ret = dhcpc_request(g_dhcp_handle, &state);
+	if (ret != OK) {
+		dhcpc_close(g_dhcp_handle);
+		return WIFI_UTILS_FAIL;
+	}
+
+	netlib_set_ipv4addr(CTRL_IFNAME, &state.ipaddr);
+	netlib_set_ipv4netmask(CTRL_IFNAME, &state.netmask);
+	netlib_set_dripv4addr(CTRL_IFNAME, &state.default_router);
+
+	nvdbg("IP address : %s ----\n", inet_ntoa(state.ipaddr));
+	
+	return WIFI_UTILS_SUCCESS;
+}
+
+static wifi_utils_result stop_dhcp_client(void)
+{
+	if (g_dhcp_handle == NULL) {
+		return WIFI_UTILS_FAIL;
+	} else {
+		dhcpc_close(g_dhcp_handle);
+		g_dhcp_handle = NULL;
+	}
+
+	return WIFI_UTILS_SUCCESS;
+}
+
+static wifi_utils_result start_dhcp_server(void)
+{
+	struct in_addr in = {.s_addr = 0x012fa8c0};
+	netlib_set_ipv4addr(CTRL_IFNAME, &in);
+	in.s_addr = 0x00ffffff;
+	netlib_set_ipv4netmask(CTRL_IFNAME, &in);
+	in.s_addr = 0x012fa8c0;
+	netlib_set_dripv4addr(CTRL_IFNAME, &in);
+	
+	if (dhcpd_start(CTRL_IFNAME, g_new_sta_join) < 0) {
+		ndbg("DHCP Server - started fail\n");
+		return WIFI_UTILS_FAIL;
+	}
+
+	nvdbg("DHCP Server - started success\n");
+	return WIFI_UTILS_SUCCESS;
+}
+
+static wifi_utils_result stop_dhcp_server(void)
+{
+	struct in_addr in = {.s_addr = INADDR_NONE};
+	netlib_set_ipv4addr(CTRL_IFNAME, &in);
+	netlib_set_ipv4netmask(CTRL_IFNAME, &in);
+	netlib_set_dripv4addr(CTRL_IFNAME, &in);
+	
+	dhcpd_stop();
+
+	return WIFI_UTILS_SUCCESS;
+}
+
+static void wifi_status_set(connect_status status)
+{
+#ifdef CONFIG_DEBUG_NET_INFO
+	static char connect_status_str[5][25] =
+	{
+			// STA mode
+			"AP_DISCONNECTED",
+			"AP_CONNECTED",
+
+			// SOFTAP mode
+			"CLIENT_CONNECTED",
+			"CLIENT_DISCONNECTED"
+	};
+#endif
+
+	if(g_manager_info.status != status)
+	{
+		nvdbg("Wifi Network Status Changes from [%s] to [%s]",
+				connect_status_str[g_manager_info.status], connect_status_str[status]);
+		g_manager_info.status = status;
+	}
+}
+
+static void ap_connect_event_func(void)
+{
+	wifi_manager_cb *wifi_cb = g_manager_info.wmcb;
+	if(g_manager_info.mode == STA_MODE)
```

Comment:
1. need space after if
```
if(xx) -> if (xx)
```
2. location of opening bracket
```
if (xx) {
```

Reply:
I did go through fomatter.sh